File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/expressions/And.java
package com.example.robotkarolar.karollogic.instructions.expressions;

import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor;

public class And extends Expression {

    private Expression left;
    private Expression right;

    public And(Expression left, Expression right) {
        this.left = left;
        this.right = right;
    }

    public void setLeft(Expression left) {
        this.left = left;
    }

    public void setRight(Expression right) {
        this.right = right;
    }

    public Expression getLeft() {
        return left;
    }

    public Expression getRight() {
        return right;
    }

    @Override
    public void accept(InstructionVisitor visitor) {
        visitor.accept(this);
    }

    @Override
    public boolean accept(ExpressionVisitor visitor) {
        return visitor.accept(this);
    }
}

File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/expressions/And.kt
package com.example.robotkarolar.karollogic.instructions.expressions

import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor
import kotlinx.parcelize.Parcelize

@Parcelize
class And(var left: Expression, var right: Expression) : Expression() {
    override fun accept(visitor: InstructionVisitor?) {
        visitor!!.accept(this)
    }

    override fun accept(visitor: ExpressionVisitor): Boolean {
        return visitor.accept(this)
    }
}